### PR TITLE
Prepare 1.4.0

### PR DESCRIPTION
**Pull request by jneira**
* Added instructions about the automated hackage release
* Initial list of prs in the changelog
* Make progress towards #2176

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2182"></a>

### CODE REVIEW

**Review by pepeiborra — APPROVED** [review]

File: ChangeLog.md [hunk 1]
```diff
@@ -160,15 +273,15 @@ We have finally released a new version of Haskell Language Server!
 Thanks for all contributors, many bugs has been fixed, and many features has landed.
 Here are the summary of changes:
 
-- Basic support for GHC 9.0.1 is added.  
+- Basic support for GHC 9.0.1 is added.
```
Comment (Ailrun) [inline ChangeLog.md:276]: These whitespaces should not be removed, as they are parts of formatting. (they means that the following text should be the separate line, not consecutive text.

File: ChangeLog.md [hunk 2]
```diff
@@ -23,7 +136,7 @@ and some new interesting features. Here is the brief summary of changes:
   - Wingman now gives a warning if it ran out of gas during "attempt to fill hole".
   - Metaprogramming for Wingman has been improved with symbolic-name support and the `pointwise` combinator.
   - An option to enable/disable Wingman's proof state styling is added.
-  - Hole fit suggestions are now disabled for performance reasons when using Wingman.
```
Comment (Ailrun) [inline ChangeLog.md:26]: This one is good though (one whitespace does not have any special effect in markdown AFAIK)

File: ChangeLog.md (hunk 1 above)
Comment (jneira) [reply ChangeLog.md:276]: thanks for noting it, i have the editor configured to remove trailing whitespaces on save though, will turn off for this

**Comment by jneira** [comment]
@isovector feel free to extend the release description with more details about wingman changes

(same for others participants and maintainers of course :slightly_smiling_face: )

File: ChangeLog.md (hunk 1 above)
Comment (jneira) [reply ChangeLog.md]: @Ailrun done!

**Comment by jneira** [comment]
This still needs version bumps

**Comment by jneira** [comment]
> 
> 
> This still needs version bumps

Done, all minor revisiones, although technically ghcide should be a major one as a public definition was removed.
All but wingman, lot of changes to verify so i bump up to 1.4.0 (//cc @isovector)

**Comment by jneira** [comment]
Will label it for merge tomorrow if nobody disagree

File: ChangeLog.md [hunk 3]
```diff
@@ -1,5 +1,151 @@
...
+- We have a *new brand plugin to let you rename indentifiers* thanks to @OliverMadine's great work
+![rename-demo](https://user-images.githubusercontent.com/30090176/133071923-b3d91b2a-619f-4151-ab76-ca50442e5ab4.gif)
+- *Call hierarchy plugin has been improved* thanks to @July541:
+  - Add call from type signature
+  - Add call from a function pattern
+  - Go to typeclass instance directly
+- We have new docs thanks to @michaelpj: <https://haskell-language-server.readthedocs.io>
+- Now you can ask the executable for included plugins with: `haskell-language-server --list-plugins`
+- *DEPRECATION NOTICE*: after this release we will drop support for ghc versions: 8.6.4, 8.10.3 and 8.10.4
+  - The advise is upgrade ghc to the last minor version: 8.6.5 or 8.10.7
```
Comment (Ailrun) [inline ChangeLog.md:18]: How about moving this notice to above so that it is easy-to-find?

Comment (jneira) [reply ChangeLog.md:18]: Hmm yeah it needs more visibility but i think it is better to start with good news, what about give it a dedicated header?

**Comment by isovector** [comment]
i am happy to write release notes tomorrow morning if it's not too late

File: ChangeLog.md (hunk 3 above)
Comment (Ailrun) [reply ChangeLog.md:18]: Sounds good for me

**Comment by jneira** [comment]
> i am happy to write wingman release notes tomorrow morning if it's not too late

@isovector 
I've just write down a dedicated section with the highlights, feel free to edit or replace it

File: ChangeLog.md (hunk 3 above)
Comment (jneira) [reply ChangeLog.md]: done!

**Comment by jneira** [comment]
> 
> 
> i am happy to write wingman release notes tomorrow morning if it's not too late

If you dont mind i am gonna label as ready to merge, we can add modifications in new prs (or even update the release if it is done)

File: ChangeLog.md [hunk 4]
```diff
@@ -1,5 +1,162 @@
 # Changelog for haskell-language-server
 
+## 1.4.0
+
+After a month of vacation a new hls release has arrived:
+
+- Support for ghc 8.10.6 and 8.10.7
+- The ormolu formatter plugin works with ghc 9.0.1
+- We have a *new brand plugin to let you rename indentifiers* thanks to @OliverMadine's great work
+![rename-demo](https://user-images.githubusercontent.com/30090176/133071923-b3d91b2a-619f-4151-ab76-ca50442e5ab4.gif)
```
Comment (pepeiborra) [inline ChangeLog.md:10]: The plugin is disabled in Cabal so it won't be possible to enable it for binary users. This needs to be highlighted

**Review by pepeiborra — CHANGES_REQUESTED** [review]

File: ChangeLog.md (hunk 4 above)
Comment (jneira) [reply ChangeLog.md:10]: oh, didnt notice that :worried:

Comment (jneira) [reply ChangeLog.md:10]: So i think it is better to remove it from the description directly, as advanced users enabling it on purpose already are aware

Comment (jneira) [reply ChangeLog.md]: Removed and opened #2193

Comment (jneira) [reply ChangeLog.md]: @pepeiborra are you fine with the remove?

Comment (pepeiborra) [reply ChangeLog.md]: Sure, I respect @OliverMadine choice

Comment (OliverMadine) [reply ChangeLog.md]: It seems reasonable to me to remove it until we have the multi-component support.

Comment (jneira) [reply ChangeLog.md]: ok so it only needs a stamp to go ☺️

Comment (jneira) [reply ChangeLog.md]: @pepeiborra does it needs some other change?

**Review by pepeiborra — APPROVED** [review]

**Pull request merged**